File: Frontend/src/models/game.js
class Game {
    id;
    numberOfPlayers;
    turnDuration;
    pointsToWin;
    status;
    host;



    constructor(id, numberOfPlayers, turnDuration, pointsToWin,status,host) {
        this.id = id;
        this.numberOfPlayers = numberOfPlayers;
        this.turnDuration = turnDuration;
        this.pointsToWin = pointsToWin;
        this.status=status;
        this.host=host;
    }

    static copyConstructor(game){
        if(game == null) return null;

        const copy = new Game(
            game.id,
            game.numberOfPlayers,
            game.turnDuration,
            game.pointsToWin,
            game.status,
            game.host
        );

        return copy;
    }

    static createGame(id, numberOfPlayers, turnDuration, pointsToWin, status='open', host) {
        return new Game(id, numberOfPlayers, turnDuration, pointsToWin, status, host);
    }

    open() {
        this.status = 'open';
    }

    close() {
        this.status = 'closed';
    }


}

module.exports = Game;

File: backend/src/main/java/org/example/backend/models/Game.java
package org.example.backend.models;

import jakarta.persistence.*;
import java.util.List;

@Entity
@Table(name = "\"Game\"")
public class Game implements Identifiable<String> {

    @Id
    private String gameId;
    private int numberOfPlayers;
    private int turnDuration;
    private int pointsToWin;
    private Status status;

    @OneToMany(mappedBy = "game")
    private List<Player> players;

    @ManyToOne
    @JoinColumn(name = "master_id", referencedColumnName = "userId")
    private User host;

    // Possible values for status
    public static final String STATUS_OPEN = "open";
    public static final String STATUS_CLOSED = "closed";

    public enum Status {
        open,
        ongoing,
        closed
    }

    public Game(String gameId, int numberOfPlayers, int turnDuration, int pointsToWin, Status status, User host) {
        this.gameId = gameId;
        this.numberOfPlayers = numberOfPlayers;
        this.turnDuration = turnDuration;
        this.pointsToWin = pointsToWin;
        this.status = status;
        this.host = host;
    }

    public Game() {
    }

    public String getId() {
        return gameId;
    }

    public void setId(String id) {
        this.gameId = id;
    }

    public int getNumberOfPlayers() {
        return numberOfPlayers;
    }

    public void setNumberOfPlayers(int numberOfPlayers) {
        this.numberOfPlayers = numberOfPlayers;
    }

    public int getTurnDuration() {
        return turnDuration;
    }

    public void setTurnDuration(int turnDuration) {
        this.turnDuration = turnDuration;
    }

    public int getPointsToWin() {
        return pointsToWin;
    }

    public void setPointsToWin(int pointsToWin) {
        this.pointsToWin = pointsToWin;
    }

    public Status getStatus() {
        return status;
    }

    public void setStatus(Status status) {
        this.status = status;
    }

    public User getHost() {
        return host;
    }

    public void setHost(User host) {
        this.host = host;
    }
}
